scripts/build-platypus-csd3-ampere.sh

```diff
@@ -0,0 +1,257 @@
+#!/bin/bash
+#SBATCH --nodes=1
+#SBATCH --ntasks=1
+#SBATCH --time=04:00:00
+#SBATCH --mail-type=none
+#SBATCH -p ampere
+#SBATCH -A ukaea-ap001-GPU
+#SBATCH --cpus-per-task=32
+#SBATCH --gres=gpu:1
+#SBATCH --output=platypus_gpu_build.%j.out
+#SBATCH --error=platypus_gpu_build.%j.err
+
+## WARNING: THIS SCRIPT WILL UNINSTALL ALL SPACK MODULES ASSOCIATED WITH
+## THE ARCHITECTURE DEFINED IN THE ARCH VARIABLE. IF YOU DO NOT WISH TO DO
+## THAT, COMMENT OUT THE SPACK UNINSTALL LINE BEFORE SUBMITTING THE SCRIPT
+## -> UNINSTALL LINE IN THE install_spack_deps() FUNCTION
+ARCH="linux-rocky8-zen"
+
+export compile_cores=32
+
+load_modules() {
+
+    # Load modules
+
+    # shellcheck source=/dev/null
+    . /etc/profile.d/modules.sh                # Leave this line (enables the module command)
+    module purge
+    module load rhel8/slurm
+    module use /usr/local/software/spack/spack-modules/rocky8-a100-20230831/linux-rocky8-zen3
+    module load cuda/11.7.1
+
+}
+
+set_paths() {
+
+    USER=$(whoami)
+    BUILD_PREFIX=platypus_gpu
+    BUILD_DIR_NAME=${BUILD_PREFIX}_build
+
+    ROOT_PATH=/home/${USER}/rds/rds-ukaea-ap001/${USER}
+    BUILD_PATH=${ROOT_PATH}/${BUILD_DIR_NAME}
+
+    echo "Building in ${BUILD_PATH}"
+    mkdir -p "${BUILD_PATH}" || {
+                                    echo "Failed to create ${BUILD_PATH}"
+                                                                            exit 1
+    }
+
+    cd "${BUILD_PATH}" || exit 1
+
+}
+
+check_spack() {
+
+    cd "${ROOT_PATH}" || exit 1
+
+    if [ "$(command -v spack)" ]; then
+        echo "Spack command detected. Using pre-loaded spack."
+    elif [ -f "${ROOT_PATH}"/spack/share/spack/setup-env.sh ]; then
+        echo "Spack detected in root directory. Loading."
+        # shellcheck source=/dev/null
+        . spack/share/spack/setup-env.sh
+    else
+        echo "No spack detected. Building from source."
+        git clone --depth=100 https://github.com/spack/spack.git
+        # shellcheck source=/dev/null
+        . spack/share/spack/setup-env.sh
+    fi
+
+}
+
+install_spack_deps() {
+
+    # Cleaning up everything to start with a new environment
+    spack uninstall -ay arch=${ARCH}
+    spack clean -ab
+
+    spack external find cuda@11.7.1
```

Comment:
This failed for me - although `spack external find cuda` worked; looks like the version it picked up was 11.7.99 instead. Built OK with 11.7.99 though.